beacon-chain/blockchain/block_processing_test.go

```diff
@@ -128,6 +128,7 @@ func TestReceiveBlock_ProcessCorrectly(t *testing.T) {
 		ParentRoot: genesis.ParentRoot,
 		BodyRoot:   bodyRoot[:],
 	}
+	beaconState.Eth1DepositIndex = 0
```

Comment:
Why not set this in `setupBeaconChain`?

Reply:
We don't have a good test set up function for it. `GenesisBeaconState` is used for production and it increases `Eth1DepositIndex`. To test correctly this receive block function, we have to reset `Eth1DepositIndex` to 0

Happy to set up test helper for this or we can push it to later